zrythm: init at 1.0.0-alpha.26.0.13

tshaynik opened the pull request:
Add a package for the zrythm Digital Audio Workstation.

Closes #98871

Built on work done by @andreasfelix on #129457 but adds additional
missing dependencies, uses the latest zrythm
version, makes use of dependencies that have since been added to
nixpkgs, and resolves the issue of the crash on start that previous
packaging attempt was encountering.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Add a package for the zrythm Digital Audio Workstation.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

magnetophon reviewed (approved):
LGTM!
Builds and runs here.

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -11639,6 +11639,12 @@
     email = "4ZKTUB6TEP74PYJOPWIR013S2AV29YUBW5F9ZH2F4D5UMJUJ6S@hash.domains";
     name = "Thomas Strobel";
   };
+  tshaynik = {
```
Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -0,0 +1,175 @@
+, zstd
+}:
+
+let
+  pname = "zrythm";
+  version = "1.0.0-alpha.26.0.13";
+in
+stdenv.mkDerivation {
+  inherit pname;
+  inherit version;
```
```suggestion
stdenv.mkDerivation rec {
  pname = "zrythm";
  version = "1.0.0-alpha.26.0.13";```

tshaynik replied:
Is there a reason that this is preferred? It seems like they're equivalent, and I've seen `rec` referred to as an anti-pattern (i.e. https://nix.dev/anti-patterns/language#rec-expression)
I'm happy to change it though.

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -0,0 +1,175 @@
+  inherit version;
+
+  nativeBuildInputs = [
+        git
```
Why do we need git? We probably need to patch the usage.

tshaynik replied:
Removed git, and it didn't break. Thanks!

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -0,0 +1,175 @@
+        zstd
+      ];
+
+  src = fetchFromGitHub {
```
src goes before any buildInputs.

tshaynik replied:
fixed!

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -0,0 +1,175 @@
+  meta = with lib; {
+    homepage = "https://www.zrythm.org";
+    description = "highly automated and intuitive digital audio workstation";
+    maintainers = [ maintainers.tshaynik maintainers.magnetophon ];
```
```suggestion
    maintainers = with maintainers; [ tshaynik magnetophon ];
```

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -0,0 +1,175 @@
+
+  meta = with lib; {
+    homepage = "https://www.zrythm.org";
+    description = "highly automated and intuitive digital audio workstation";
```
```suggestion
    description = "Highly automated and intuitive digital audio workstation";
```

SuperSandro2000 reviewed (changes requested):
Please format the package file with nixpkgs-fmt.

tshaynik committed:
maintainers: add tshaynik

tshaynik committed:
zrythm: init at 1.0.0-alpha.26.0.13

Closes #98871

Built on work done by @andreasfelix on #129457 but adds additional
missing dependencies, uses the latest zrythm
version, makes use of dependencies that have since been added to
nixpkgs, and resolves the issue of the crash on start that previous
packaging attempt was encountering.

tshaynik commented:
> Please format the package file with nixpkgs-fmt.

I formatted with nixpkgs-fmt, addressed all of your points, and rebased. Let me know what you think @SuperSandro2000.
